pkg/client/gen.go

```diff
@@ -43,18 +46,8 @@ type templateValues struct {
 
 // GenerateManifest fills in a template with a Sonobuoy config
 func (c *SonobuoyClient) GenerateManifest(cfg *GenConfig) ([]byte, error) {
-	if cfg.Image != "" {
-		cfg.Config.WorkerImage = cfg.Image
-	}
-
-	if cfg.ImagePullPolicy != "" {
-		cfg.Config.ImagePullPolicy = cfg.ImagePullPolicy
-	}
-
-	if cfg.Namespace != "" {
-		cfg.Config.Namespace = cfg.Namespace
-	}
-
+	// Set defaults for the config.Config and overwrite fields with the passed in config.
+	mergo.MergeWithOverwrite(cfg.Config, config.New())
```

Comment:
Given that this was inbound from the customer I think we should write some tests to ensure behavior.